azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/WebApp.java

```diff
@@ -235,14 +239,29 @@ private Update updateAppServicePlan(Update update, AppServicePlanEntity newServi
             return update.withExistingAppServicePlan(newPlanServiceModel);
         }
 
+        private Update updateDockerConfiguration(Update update, DockerConfiguration dockerConfiguration) {
+            final com.azure.resourcemanager.appservice.models.WebApp.UpdateStages.WithStartUpCommand withStartUpCommand;
+            if (StringUtils.isAllEmpty(dockerConfiguration.getUserName(), dockerConfiguration.getPassword())) {
+                withStartUpCommand = update.withPublicDockerHubImage(dockerConfiguration.getImage());
+            } else if (StringUtils.isEmpty(dockerConfiguration.getRegistryUrl())) {
+                withStartUpCommand = update.withPrivateDockerHubImage(dockerConfiguration.getImage())
+                        .withCredentials(dockerConfiguration.getUserName(), dockerConfiguration.getPassword());
+            } else {
+                withStartUpCommand = update.withPrivateRegistryImage(dockerConfiguration.getImage(), dockerConfiguration.getRegistryUrl())
+                        .withCredentials(dockerConfiguration.getUserName(), dockerConfiguration.getPassword());
+            }
+            modified = true;
+            return withStartUpCommand.withStartUpCommand(dockerConfiguration.getStartUpCommand());
```

Comment:
really odd withStartUpCommand.withStartUpCommand, please change var name of 'withStartUpCommand'